src/IO/H5/VolumeData.cpp

```diff
@@ -205,6 +207,98 @@ void VolumeData::write_volume_data(
               append_element_extents_and_connectivity(
                   &total_extents, &total_connectivity, &total_points_so_far,
                   dim, element);
+
+              // If element is 2D and the bases are both SphericalHarmonic,
+              // then add extra connections to close the surface.
+              if (dim == 2) {
+                if (element.basis[0] == Spectral::Basis::SphericalHarmonic and
+                    element.basis[1] == Spectral::Basis::SphericalHarmonic) {
+                  // Extens are (l+1, 2l+1)
+                  const size_t l = element.extents[0] - 1;
+
+                  // Connect max(phi) and min(phi) by adding more quads
+                  // to total_connectivity
+                  for (size_t j = 0; j < l; ++j) {
+                    total_connectivity.push_back(j);
+                    total_connectivity.push_back(j + 1);
+                    total_connectivity.push_back(2 * l * (l + 1) + j + 1);
+                    total_connectivity.push_back((2 * l) * (l + 1) + j);
+                  }
+
+                  // Add a new connectivity output for filling the poles
+                  // First, get the points at min(theta), which define the
+                  // boundary of the top pole to fill, and the points at
+                  // max(theta), which define the boundary of the bottom
+                  // pole to fill. Note: points are stored with theta
+                  // varying faster than phi.
+                  std::vector<int> top_pole_points{};
+                  std::vector<int> bottom_pole_points{};
+                  for (size_t k = 0; k < (l + 1) * (2 * l + 1); ++k) {
```

Comment:
I find this loop very confusing to parse. Could this be done in a more intuitive way? If I've understood correctly the first `if` equivalent to:
```
for (size_t k = 0; k < (2*l+1); ++k) {
  top_pole_points.push_back(k * (l + 1));
}
```
which I find much easier to read. If this isn't the equivalent loop, then I think some more explanation of the code is necessary. I expect the second `if` can be rewritten in a similar manner.

Reply:
Ah okay, that all makes sense now. Thank you!